src/protobuf-net.Test/Issues/Issue323.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Xunit;
+using ProtoBuf.Meta;
+
+namespace ProtoBuf.Test.Issues
+{
+    public class Issue323
+    {
+        const string ExpectedProto = @"syntax = ""proto3"";
+package ProtoBuf.Test.Issues;
+
+enum TokenType {
+   TokenType_Temporary = 0;
+   TokenType_Persistent = 1;
+}
+";
+
+        [Fact]
+        public void ProtoIncludesEnumNamePrefix()
+        {
+            string proto = Serializer.GetProto(new SchemaGenerationOptions { Types = { typeof(TokenType) },
+                Syntax = ProtoSyntax.Proto3, Flags = SchemaGenerationFlags.IncludeEnumNamePrefix });
+
+            Assert.Equal(ExpectedProto, proto, ignoreLineEndingDifferences: true);
+        }
+
+        [ProtoContract]
```

Comment:
can we add a second enum here with a `[ProtoContract(Name = "SomethingElse")]` to check that the generated prefix is `SomethingElse_` ?